dss/config.py

```diff
@@ -97,20 +99,57 @@ def get_gs_bucket() -> str:
         return Config._GS_BUCKET
 
     @staticmethod
-    def _clear_cached_config():
+    def get_allowed_email_domains() -> str:
+        if Config._ALLOWED_EMAILS is None:
+            if Config._CURRENT_CONFIG == DeploymentStage.NORMAL:
+                envvar = "DSS_SUBSCRIPTION_AUTHORIZED_DOMAINS"
+            elif Config._CURRENT_CONFIG == DeploymentStage.TEST:
+                envvar = "DSS_SUBSCRIPTION_AUTHORIZED_DOMAINS_TEST"
+            elif Config._CURRENT_CONFIG == DeploymentStage.TEST_FIXTURE:
+                envvar = "DSS_SUBSCRIPTION_AUTHORIZED_DOMAINS_TEST"
+            elif Config._CURRENT_CONFIG == DeploymentStage.ILLEGAL:
+                raise Exception("bucket config not set")
+
+            if envvar not in os.environ:
+                raise Exception(
+                    f"Please set the {envvar} environment variable")
+            Config._ALLOWED_EMAILS = os.environ[envvar]
+
+        return Config._ALLOWED_EMAILS
+
+    @staticmethod
+    def _clear_cached_bucket_config():
         # clear out the cached bucket settings.
         Config._S3_BUCKET = None
         Config._GS_BUCKET = None
 
+    @staticmethod
+    def _clear_cached_email_config():
+        # clear out the cached email settings.
+        Config._ALLOWED_EMAILS = None
+
 
 @contextmanager
 def override_bucket_config(temp_config: DeploymentStage):
     original_config = Config._CURRENT_CONFIG
-    Config._clear_cached_config()
+    Config._clear_cached_bucket_config()
+
+    try:
+        Config._CURRENT_CONFIG = temp_config
+        yield
+    finally:
+        Config._CURRENT_CONFIG = original_config
+        Config._clear_cached_bucket_config()
+
+
+@contextmanager
+def override_email_config(temp_config: DeploymentStage):
```

Comment:
This doesn't appear to be used / necessary any more.  Can this be nuked?

Alternatively, could this be used to fake the 403 instead of the flag?